### PR TITLE
Add CmdFlip and CmdRoll

### PR DESCRIPTION
**Pull request by Legomenon-gh**
This allows to support randomizing outcomes within a game for tournament support, this is likely a temporary solution until bot
support is available within a game lobby.

Addresses https://github.com/ZeroK-RTS/Zero-K-Infrastructure/issues/2805

### CODE REVIEW

**Comment by Legomenon-gh** [comment]
I'm not wed to having both flip and roll available and can delete one of the two if that's necessary (I'd have put the coin flip behavior as the `!roll` default if rolling a coin made sense in English).

Obviously recreating a Random instance every time isn't cryptographically secure, but that's not a problem here (map selection does the same thing).

File: ZkLobbyServer/autohost/Commands/CmdRoll.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            if (int.TryParse(arguments, out maximum)) return string.Empty;
+            else return null;
+        }
+
+        public override async Task ExecuteArmed(ServerBattle battle, Say e)
+        {
+            if (maximum >= 1)
+            {
+                var result = new Random().Next(1, maximum);
```
Comment (DeinFreund) [inline ZkLobbyServer/autohost/Commands/CmdRoll.cs:27]: maximum + 1?

File: ZkLobbyServer/autohost/Commands/CmdFlip.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using System.Threading.Tasks;
+using LobbyClient;
+
+namespace ZkLobbyServer
+{
+    public class CmdFlip : BattleCommand
+    {
+        public override string Help => "Flips a coin";
+        public override string Shortcut => "flip";
+        public override AccessType Access => AccessType.NotIngame;
```
Comment (DeinFreund) [inline ZkLobbyServer/autohost/Commands/CmdFlip.cs:11]: why not anywhere?

File: ZkLobbyServer/autohost/Commands/CmdRoll.cs (hunk 1 above)
Comment (Legomenon-gh) [reply ZkLobbyServer/autohost/Commands/CmdRoll.cs:27]: Sigh, thanks, I forgot to update this when I moved to not-zero-indexing. Very dumb...

File: ZkLobbyServer/autohost/Commands/CmdFlip.cs (hunk 2 above)
Comment (Legomenon-gh) [reply ZkLobbyServer/autohost/Commands/CmdFlip.cs:11]: I can make it everywhere if you want, I can't see a good reason to have random numbers while ingame (I could see some funny users spamming for random numbers as a taunt, but trolls gonna troll anyway)

**Review by DeinFreund — APPROVED** [review]

**Pull request merged**